src/engine/SCons/SConf.py

```diff
@@ -592,8 +593,30 @@ def TryBuild(self, builder, text = None, extension = ""):
             raise SCons.Errors.UserError('Missing SPAWN construction variable.')
 
         nodesToBeBuilt = []
+        sourcetext = self.env.Value(text)
+        f = "conftest"
+
+        if text is not None:
+            textSig = SCons.Util.MD5signature(sourcetext)
+            textSigCounter = str(_ac_build_counter[textSig])
+            _ac_build_counter[textSig] += 1
```

Comment:
I'd been hoping to entirely eliminate the counter. But without it, you end up with many configure tests which have the same text body (notably, the empty body). If those configure tests came from different environments, you get the dreaded "scons: warning: Two different environments were specified for target" issue. So, we re-purpose the counter to differentiate identical files. That is sort of unfortunate, but it still works, because if the numbering ever changes in a "bad" way, it shouldn't matter: the sources are always identical.

Reply:
Perhaps the answer then not to have different source filenames, but different target file names. This would avoid `"scons: warning: Two different environments were specified for target"`

Reply:
@bdbaddog - We collide on the `target` nodes of `SConfSourceBuilder`, so that is what this is actually doing: producing different target names.